### PR TITLE
Save the Date - Static Resource Load

### PR DESCRIPTION
**Pull request by Suhail6inkling**
---
name: Save the Date fix
about: Move the loading of the static JSON in the Save the Date cog
issue: #147 

---

Solves this issue by moving the loading of the JSON file into the module namespace, so that it is not loaded every time the command is called

Hope this is okay :D

### CODE REVIEW

**Comment by jchristgit** [comment]
I think the most preferable solution would be only loading the resource when the *cog* is actually loaded, so loading it in the `setup` function and passing it to the class. Curious what the others think.

**Comment by sco1** [comment]
Is there a functional difference between when it's loaded (module vs. init vs. setup) with how we're using the cogs?

Regardless, it would probably be handy to settle on one and enforce it across the org since they're pretty scattered.

**Comment by MarkKoz** [comment]
I think it would only have different behaviour when a cog is reloaded. I can't confirm that though.

**Comment by scragly** [comment]
There would be no functional difference between placing it in setup vs cog init, and only the difference of referring to it via `self` compared to defining it on the module-level.

When reloading an extension, the entire extension (full package if package-style, full module if module-style) will be unloaded, reimported and the setup run again. Due to this, there's no functional difference for reload on any of the options.

The standard thing to do here is to put it on the module level, keeping it in full namespace throughout the module. I personally feel doing it any other way is likely just thinking too much about it.

File: bot/seasons/valentines/savethedate.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 HEART_EMOJIS = [":heart:", ":gift_heart:", ":revolving_hearts:", ":sparkling_heart:", ":two_hearts:"]
 
+with open(Path('bot', 'resources', 'valentines', 'date_ideas.json'), 'r', encoding="utf8") as f:
+    valentine_dates = load(f)
```
Comment (scragly) [inline bot/seasons/valentines/savethedate.py]:
```suggestion
    VALENTINE_DATES = load(f)
```

File: bot/seasons/valentines/savethedate.py [hunk 2]
```diff
@@ -26,17 +29,15 @@ def __init__(self, bot):
...
-            random_date = random.choice(valentine_dates['ideas'])
-            emoji_1 = random.choice(HEART_EMOJIS)
-            emoji_2 = random.choice(HEART_EMOJIS)
-            embed = discord.Embed(
-                title=f"{emoji_1}{random_date['name']}{emoji_2}",
-                description=f"{random_date['description']}",
-                colour=Colours.pink
-            )
-            await ctx.send(embed=embed)
+        random_date = random.choice(valentine_dates['ideas'])
```
Comment (scragly) [inline bot/seasons/valentines/savethedate.py]:
```suggestion
        random_date = random.choice(VALENTINE_DATES['ideas'])
```

**Pull request merged**